### PR TITLE
Feature/product page be

### LINKED ISSUE #9: User can see their product history
As a User, when I visit my profile page, I can see a history of products that have been delivered to me and what I have kept.

### PR DESCRIPTION
**Pull request by javier-aguilar**
### Type of change made:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling (Styling to new or existing files)
- [ ] Testing

### Detailed Description
Added user relation to product

### Why is this change required? What problem does it solve?
You can now query for products related to a given user

### Was any previously written code changed?
Yes

### Directory modified:
- Update User Model
- Update Product Model
- Add api/src/modules/userproduct

### CODE REVIEW

**Review by maddyrandle — APPROVED** [review]
Thanks for all of your hard work. This is awesome!

File: code/api/src/modules/user/resolvers.js [hunk 1]
```diff
@@ -60,7 +60,14 @@ export async function login(parentValue, { email, password }) {
 
 // Get by ID
 export async function getById(parentValue, { id }) {
-  return await models.User.findOne({ where: { id } })
+  // const output = await models.User.findOne({ where: { id }, include: models.Product })
```
Comment (mikez321) [inline code/api/src/modules/user/resolvers.js:63]: Probably should remove these commented lines.

File: code/api/src/modules/user/types.js [hunk 2]
```diff
@@ -16,7 +18,24 @@ const UserType = new GraphQLObjectType({
     updatedAt: { type: GraphQLString },
     address: { type: GraphQLString },
     description: { type: GraphQLString },
-    image: { type: GraphQLString }
+    image: { type: GraphQLString },
+    products: { type: new GraphQLList(userProduct) }
```
Comment (mikez321) [inline code/api/src/modules/user/types.js:22]: Nice work figuring this part out!!

File: code/api/src/modules/userproduct/mutations.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      name: 'userId',
+      type: GraphQLInt
+    },
+
+    productId: {
+      name: 'productId',
+      type: GraphQLInt
+    },
+
+    kept: {
```
Comment (mikez321) [inline code/api/src/modules/userproduct/mutations.js:22]: Unfortunate how we might not get to this part... but the thought was there

File: code/api/src/modules/userproduct/query.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import UserProductType from './types'
+import { getAll, getByUser, get } from './resolvers'
+
+// userproduct by user
+export const userproductByUser = {
+  type: new GraphQLList(UserProductType),
+  resolve: getByUser
+}
+
+// UserProduct All
```
Comment (mikez321) [inline code/api/src/modules/userproduct/query.js:14]: Not sure that we need all these functions in here, might should test 'em tho.

File: code/api/src/setup/models.js [hunk 5]
```diff
@@ -8,7 +8,8 @@ const models = {
   User: databaseConnection.import('../modules/user/model'),
   Product: databaseConnection.import('../modules/product/model'),
   Crate: databaseConnection.import('../modules/crate/model'),
-  Subscription: databaseConnection.import('../modules/subscription/model')
+  Subscription: databaseConnection.import('../modules/subscription/model'),
+  UserProduct: databaseConnection.import('../modules/userproduct/model')
```
Comment (mikez321) [inline code/api/src/setup/models.js:12]: Good call on in-general cleaning this all up from how I had abstracted everything.

**Review by mikez321 — APPROVED** [review]

**Comment by mikez321** [comment]
There's a few commented lines in there but we'll be making another PR if we don't want them out right away.

**Pull request merged**